flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImpl.java

```diff
@@ -67,8 +70,12 @@ void run() {
 		} catch (Exception ex) {
 			thrown = ex;
 		} finally {
-			cleanupRequests();
-			dispatcher.fail(thrown == null ? new CancellationException() : thrown);
+			try {
+				closeAll(this::cleanupRequests, () -> dispatcher.fail(thrown == null ? new CancellationException() : thrown));
+			} catch (Exception e) {
+				//noinspection NonAtomicOperationOnVolatileField
+				thrown = ExceptionUtils.firstOrSuppressed(e, thrown);
+			}
```

Comment:
nit: This is a really hacky piece of code :)

I'm fine with keeping it as it is to not trigger another re-build over a minor issue.